sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/SqlQuery.java

```diff
@@ -45,29 +44,26 @@
     class Factory implements HandlerFactory {
         @Override
         public Handler buildHandler(Class<?> sqlObjectType, Method method) {
-            return new QueryHandler(sqlObjectType, method, ResultReturner.forMethod(sqlObjectType, method));
+            return new QueryHandler(sqlObjectType, method);
         }
     }
 
-    class QueryHandler extends CustomizingStatementHandler {
-        private final Class<?> sqlObjectType;
-        private final Method method;
+    class QueryHandler extends CustomizingStatementHandler<Query> {
         private final ResultReturner magic;
```

Comment:
I still wonder if it can be cleaned up more.  There's too many levels of indirection and change of control in the results creation process.  But it's probably better to ship this feature and do cleanup separately.